File: docs/snippets/cpu/ov_execution_mode.cpp
// Copyright (C) 2023 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include <openvino/runtime/core.hpp>

int main() {
    //! [ov:execution_mode:part0]
    ov::Core core;
    // in case of Accuracy
    core.set_property("CPU", ov::hint::execution_mode(ov::hint::ExecutionMode::ACCURACY));
    // in case of Performance
    core.set_property("CPU", ov::hint::execution_mode(ov::hint::ExecutionMode::PERFORMANCE));
    //! [ov:execution_mode:part0]

    return 0;
}

File: docs/articles_en/assets/snippets/ov_execution_mode.py
# Copyright (C) 2018-2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

#! [ov:execution_mode:part0]
import openvino as ov
import openvino.properties.hint as hints

core = ov.Core()
# in case of Accuracy
core.set_property(
    "CPU",
    {hints.execution_mode: hints.ExecutionMode.ACCURACY},
)
# in case of Performance
core.set_property(
    "CPU",
    {hints.execution_mode: hints.ExecutionMode.PERFORMANCE},
)
#! [ov:execution_mode:part0]
